sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -51,6 +51,11 @@ case class CostBasedJoinReorder(conf: CatalystConf) extends Rule[LogicalPlan] wi
 
   def reorder(plan: LogicalPlan, output: AttributeSet): LogicalPlan = {
     val (items, conditions) = extractInnerJoins(plan)
+    // Find the star schema joins. Currently, it returns the star join with the largest
+    // fact table. In the future, it can return more than one star join (e.g. F1-D1-D2
+    // and F2-D3-D4).
+    val starJoinPlans = StarSchemaDetection(conf).findStarJoins(items, conditions.toSeq)
```

Comment:
@ron8hu We already ran TPC-DS with star schema and the results are documented in the design doc. I don't think there is a question about its value. 

I am familiar with Pat Selinger's paper since I've been working in the IBM DB2 optimizer for several years. What Zhenhua and I are discussing here is how to integrate the star join plans with his new DP planning. There are no competing planning algorithm that needs to be tested.